src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+    /** The footer of the website. This can be a component. */
+    footer?: any;
+    /** The height of the header. It will not hide on overflow. */
+    headerHeight?: string | number;
+    /** The height of the footer. It will not hide on overflow. */
+    footerHeight?: string | number;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
     className?: string;
     children?: any;
 }
 
-export const Page = ({ children }: PageProps): React.ReactElement<any> => (
-    <StyledPageElement className="anchor-page">
+interface StyledLayoutProps {
+    layoutWidth?: string | number;
+    layoutBackgroundColor?: string;
+}
+
+interface StyledContentProps {
+    contentWidth?: string | number;
+    contentBackgroundColor?: string;
+}
+
+// Components
+// ------------------------------------------------------------------------------------------------------------------
+export const Page = (props: PageProps): React.ReactElement<any> => (
+    <StyledPage className={classnames('anchor-page', props.className)}>
         <NormalizeCSS />
         <GlobalCSS />
-        <Header />
-        <StyledBodyElement>{children}</StyledBodyElement>
-        <Footer />
-    </StyledPageElement>
+        <Grid
+            columns={1}
+            rows={`minmax(${props.headerHeight},auto) 1fr minmax(${
+                props.footerHeight
+            },auto)`}
+            height={'100vh'}
+        >
+            {props.header && <Cell>{props.header}</Cell>}
+
+            <Cell>{props.children}</Cell>
+
+            {props.footer && <Cell>{props.footer}</Cell>}
+        </Grid>
+    </StyledPage>
+);
+
+Page.defaultProps = PageDefaultProps;
```

Comment:
Let's talk about default props. I think we need a standard here and I wanna hear your input. I'd either like all defaults to be in the method signature, or all defaults like this. My opinion is that this isn't quite as readable for the average person, but I do wanna chat about it from other perspectives.

Reply:
I'm up for the discussion, I'm definitely not married to the way I have it.

Reply:
I'll add a mtg to discuss this. I'll also try and get a few others in the room to talk about it.